packages/presentation/lib/home_screen/screen/home_widget.dart

```diff
@@ -9,39 +10,49 @@ class HomeWidget extends StatefulWidget {
 }
 
 class _HomeWidgetState extends State<HomeWidget> {
-  final palindromeCaseImpl = PalindromeCaseImpl();
+  final palindromeBlock = PalindromeBlock(PalindromeCaseImpl());
   String textResult = "";
   String myText = "";
 
+  void _checkPalindrome() {
+    palindromeBlock.inputStreamSink.add(myText);
+  }
+
+  void _onChangedText(String text) {
+    myText = text;
+  }
+
+  @override
+  void dispose() {
+    palindromeBlock.dispose();
+    super.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
         home: Scaffold(
       body: Column(children: [
         TextField(
-            onChanged: (text) {
-              myText = text.toString();
-            },
+            onChanged: _onChangedText,
             decoration: const InputDecoration(
                 border: InputBorder.none,
                 hintText: "Введите строку",
                 fillColor: Colors.black12,
                 filled: true)),
         ElevatedButton(
-          onPressed: () {
-            setState(() {
-              final result = palindromeCaseImpl.isPalindrome(myText);
-              myText = myText.trim();
-              if (result) {
-                textResult = "Это палиндром";
-              } else {
-                textResult = "Это не палиндром";
-              }
-            });
-          },
+          onPressed: _checkPalindrome,
           child: const Text("Проверить строку"),
         ),
-        Text(textResult)
+        StreamBuilder(
+            stream: palindromeBlock.outputStreamSink,
+            builder: (context, snapshot) {
+              if (snapshot.data != null) {
+                return Text(snapshot.data as String);
+              } else {
+                return Text('');
+              }
+            })
```

Comment:
in most cases there's only one stream which is used to communicate via bloc and widget. in this case u should have the following widget hierarchy:

StreamBuilder
|
|- Scaffold

no need to change it now, but use this approach while doing a bigger project

Reply:
okay, i'll be know